src/NServiceBus.Core/Pipeline/Outgoing/AttachSenderRelatedInfoOnMessageBehavior.cs

```diff
@@ -19,6 +20,20 @@ public Task Invoke(IRoutingContext context, Func<IRoutingContext, Task> next)
             {
                 message.Headers[Headers.TimeSent] = DateTimeOffsetHelper.ToWireFormattedString(DateTimeOffset.UtcNow);
             }
+
+            if (!message.Headers.ContainsKey(Headers.DeliverAt))
```

Comment:
Why not set this? The header is prefixed `NServiceBus.` so we own it. Shouldn't the dispatch properties be the source of truth and this do whatever it says?